book/src/04_3_1_i2c.md

```diff
@@ -1,21 +1,132 @@
-## I2C Sensor Reading Exercise
+## I2C Sensor Reading Exercise 
 
-TODO `i2c-sensor-reading/exercise/src/main.rs` is an exercise skeleton that will build. The task is to use an existing driver from crates.io to read out the temperature and humidity sensor over i2c. After that, a second sensor will be read out over the same i2c bus using `shared-bus`. The driver for the second sensor is available locally in `common/`.
+In this exercise you will learn how to read out sensors on the i2c bus. 
+ 
+ 
+The board has two sensors that can be read via the I2C bus: `
 
-`i2c-sensor-reading/solution/src/main.rs` contains a working solution.
 
+| Peripheral                | Part number  | Reference                                                                   | Crate | Address |
+|---------------------------|--------------|-----------------------------------------------------------------------------|-------|---------|
+| IMU                       | ICM-42670-P  | [Datasheet](https://invensense.tdk.com/download-pdf/icm-42670-p-datasheet/) | [Link](https://crates.io/crates/icm42670)   | 0x68 |
+| Temperature and Humidity  | SHTC3        | [Datasheet](https://www.sensirion.com/en/environmental-sensors/humidity-sensors/digital-humidity-sensor-shtc3-our-new-standard-for-consumer-electronics/)            | [Link](https://crates.io/crates/shtcx)       | 0x70 |
 
-* Define two pins, one as SDA and one as SCL. Use them to instantiate an Instance of the I²C peripheral. 
 
-* Using the existing driver crates, make an instance of each sensor, passing the I²C instance into them. 
+ The task is to use an existing driver from crates.io to read out the temperature and humidity sensor over i2c. After that, a second sensor will be read out over the same i2c bus using `shared-bus`. The driver for the second sensor is available locally in `common/`.
 
+### Part 1: Reading Temperature & Humidity
 
-TODO `i2c-driver/exercise/src/main.rs` contains working code. Running it without modifying the driver file will yield errors. 
+Create an instance of the of the Humidity sensor SHTC3 and read and print the values for humidity and temperature every 600 ms.
 
-`i2c-driver/solution/src/imc42670p.rs` provides a solution to the task. 
 
 
+`i2c-sensor-reading/examples/part_1.rs` contains a working solution of Part 1.
 
+`i2c-sensor-reading/src/main.rs` contains skeleton code, that already contains necessary imports for this part. 
 
+#### Steps:
 
+✅ Go to the `i2c-sensor-reading/` folder and open the relevant documentation with the following command:
+  
+```
+$ cargo doc --open
+```
+ 
+✅ Define two pins, one as SDA and one as SCL. 
+
+| Signal     | GPIO        |
+|------------|-------------|
+| SDA        | GPIO10      |
+| SCL        | GPIO8       |
+
+✅ Create an Instance of the I²C peripheral with the help of the documentation. 
```

Comment:
Maybe some people won't know where to look exactly. One possibility would be to mention `esp_idf_hal`.

Reply:
its your own generated documentation!